### PR TITLE
Use discovery logic in establishing cache in kubernetesenv adapter

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR adds logic to use the discovery API before trying to establish Watch/List behavior on various API GroupVersions. This is required to fail gracefully in situations in which not all of the requested APIs are supported in environments that Mixer is deployed.

See: https://github.com/istio/istio/issues/6152

### CODE REVIEW

**Comment by douglas-reid** [comment]
@kyessenov does this help with your observed issues?

If so, I can clean it up and merge.

File: mixer/adapter/kubernetesenv/config/adapter.kubernetesenv.config.pb.html [hunk 1]
```diff
@@ -63,6 +63,18 @@ <h2 id="Params">Params</h2>
 
 <p>Default: 5 minutes</p>
 
+</td>
+</tr>
+<tr id="Params.is_v18_cluster">
+<td><code>isV18Cluster</code></td>
+<td><code>bool</code></td>
+<td>
+<p>Provides support for kubernetes clusters from v1.8.x vintage. Set this to
```
Comment (kyessenov) [inline mixer/adapter/kubernetesenv/config/adapter.kubernetesenv.config.pb.html]: I don't think we should track k8s version. API groups are independently versioned, and are sometimes packaged differently for distributions of k8s. It is not guaranteed a certain version of k8s has or has not a particular API version, I believe.

**Comment by kyessenov** [comment]
Let's just make the APA fail gracefully. You could try to list a resource declaration before using it, to verify whether it's available or not.

**Comment by douglas-reid** [comment]
@kyessenov agreed. just wanted to throw this across and see if it fixed your issue first. 

Added discovery logic to the code. PTAL.

**Comment by douglas-reid** [comment]
https://github.com/istio/istio/issues/7675#issuecomment-415447894 has me concerned that this PR should be reopened.

**Comment by douglas-reid** [comment]
Well, reopening didn't seem to work... and I cannot reclose. Wonder what happens if I open a second PR.

**Pull request closed without merging**